packages/application-shell/src/components/fetch-project/fetch-project.js

```diff
@@ -145,53 +145,48 @@ export const mapAllDataFencesToGroupedObjectShape = allAppliedDataFences => {
   );
 };
 
-class FetchProject extends React.Component {
-  static displayName = 'FetchProject';
-  static propTypes = {
-    projectKey: PropTypes.string,
-    skip: PropTypes.bool,
-    children: PropTypes.func.isRequired,
-  };
-  static defaultProps = {
-    skip: false,
-  };
-  shouldComponentUpdate(nextProps) {
-    return !deepEqual(this.props, nextProps);
-  }
-  render() {
-    return (
-      <Query
-        query={ProjectQuery}
-        variables={{
-          target: GRAPHQL_TARGETS.MERCHANT_CENTER_BACKEND,
-          projectKey: this.props.projectKey,
-        }}
-        skip={this.props.skip}
-      >
-        {({ data, loading, error }) =>
-          this.props.children({
-            isLoading: loading,
-            error,
-            project: data &&
-              data.project && {
-                ...data.project,
-                permissions: mapAllAppliedToObjectShape(
-                  data.project.allAppliedPermissions
-                ),
-                dataFences: mapAllDataFencesToGroupedObjectShape(
-                  data.project.allAppliedDataFences
-                ),
-                actionRights: mapAllAppliedToGroupedObjectShape(
-                  data.project.allAppliedActionRights
-                ),
-                menuVisibilities: mapAllAppliedToObjectShape(
-                  data.project.allAppliedMenuVisibilities
-                ),
-              },
-          })
-        }
-      </Query>
-    );
-  }
-}
+const FetchProject = props => {
+  const { loading, data, error } = useQuery(ProjectQuery, {
+    onError: reportErrorToSentry,
```

Comment:
In AppShell we don't explicitly handle any `error`, so we report it to sentry in case we get some.

Reply:
We didn't do this before though, right? Does Apollo maybe have a more generic way to do this e.g. in a link? So we have a SentryErrorReportingLink which enables us having all errors also from the app to be reported? Can be a follow up I just scratched my head cause we added this here now.

Reply:
Yeah that’s one of those “uncertain topics” that we have. In general we don’t always handle those kind of errors, at least not always and not consistently.
We could have an error link I suppose yes, although I’m not sure we want to report all errors. I think I would prefer to do this for now, on a query-by-query basis.

PS: I added this here because I found out or realized that while I was working on this PR.

Reply:
Sure. I am not generally arguing against it. Just saying that maybe we can encapsulate it more and offer it through a link. Maybe an opt-in or opt-out thing through a reportError prop on each query or mutation.